### PR TITLE
new rule added to 0020-syslog_rules.xml

### PR DESCRIPTION
**Pull request by SitoRBJ**
We have created a new rule regarding sudo-related actions.

The goal is to have a rule that generates an alert when a user other than root successfully performs sudo-related actions.

The new rule is:

```
  <rule id="5407" level="3">
    <if_sid>5400</if_sid>
    <regex> ; USER=\S+ ; COMMAND=| ; USER=\S+ ; TSID=\S+ ; COMMAND=</regex>
    <description>Successful sudo executed </description>
    <group>pci_dss_10.2.5,pci_dss_10.2.2,gpg13_7.6,gpg13_7.8,gpg13_7.13,gdpr_IV_32.2,</group>
  </rule>
```

Logtest output:

```
2017-12-28T10:37:47.968126+00:00 lognode-01.local sudo: vagrant : TTY=pts/0 ; PWD=/home/vagrant ; USER=ossec ; COMMAND=/bin/id


**Phase 1: Completed pre-decoding.
       full event: '2017-12-28T10:37:47.968126+00:00 lognode-01.local sudo: vagrant : TTY=pts/0 ; PWD=/home/vagrant ; USER=ossec ; COMMAND=/bin/id'
       timestamp: '2017-12-28T10:37:47.968126+00:0'
       hostname: 'lognode-01.local'
       program_name: 'sudo'
       log: 'vagrant : TTY=pts/0 ; PWD=/home/vagrant ; USER=ossec ; COMMAND=/bin/id'

**Phase 2: Completed decoding.
       decoder: 'sudo'
       srcuser: 'vagrant'
       tty: 'pts/0'
       pwd: '/home/vagrant'
       dstuser: 'ossec'
       command: '/bin/id'

**Phase 3: Completed filtering (rules).
       Rule id: '5403'
       Level: '4'
       Description: 'First time user executed sudo.'
**Alert to be generated.


2017-12-28T10:37:47.968126+00:00 lognode-01.local sudo: vagrant : TTY=pts/0 ; PWD=/home/vagrant ; USER=ossec ; COMMAND=/bin/id


**Phase 1: Completed pre-decoding.
       full event: '2017-12-28T10:37:47.968126+00:00 lognode-01.local sudo: vagrant : TTY=pts/0 ; PWD=/home/vagrant ; USER=ossec ; COMMAND=/bin/id'
       timestamp: '2017-12-28T10:37:47.968126+00:0'
       hostname: 'lognode-01.local'
       program_name: 'sudo'
       log: 'vagrant : TTY=pts/0 ; PWD=/home/vagrant ; USER=ossec ; COMMAND=/bin/id'

**Phase 2: Completed decoding.
       decoder: 'sudo'
       srcuser: 'vagrant'
       tty: 'pts/0'
       pwd: '/home/vagrant'
       dstuser: 'ossec'
       command: '/bin/id'

**Phase 3: Completed filtering (rules).
       Rule id: '5407'
       Level: '3'
       Description: 'Successful sudo executed '
**Alert to be generated.
```

If the user would be root:

```
Jun 26 17:21:43 manager sudo:  manager : TTY=pts/0 ; PWD=/home/manager ; USER=root ; COMMAND=/bin/su"


**Phase 1: Completed pre-decoding.
       full event: 'Jun 26 17:21:43 manager sudo:  manager : TTY=pts/0 ; PWD=/home/manager ; USER=root ; COMMAND=/bin/su"'
       timestamp: 'Jun 26 17:21:43'
       hostname: 'manager'
       program_name: 'sudo'
       log: ' manager : TTY=pts/0 ; PWD=/home/manager ; USER=root ; COMMAND=/bin/su"'

**Phase 2: Completed decoding.
       decoder: 'sudo'
       srcuser: 'manager'
       tty: 'pts/0'
       pwd: '/home/manager'
       dstuser: 'root'
       command: '/bin/su"'

**Phase 3: Completed filtering (rules).
       Rule id: '5403'
       Level: '4'
       Description: 'First time user executed sudo.'
**Alert to be generated.


Jun 26 17:21:43 manager sudo:  manager : TTY=pts/0 ; PWD=/home/manager ; USER=root ; COMMAND=/bin/su"


**Phase 1: Completed pre-decoding.
       full event: 'Jun 26 17:21:43 manager sudo:  manager : TTY=pts/0 ; PWD=/home/manager ; USER=root ; COMMAND=/bin/su"'
       timestamp: 'Jun 26 17:21:43'
       hostname: 'manager'
       program_name: 'sudo'
       log: ' manager : TTY=pts/0 ; PWD=/home/manager ; USER=root ; COMMAND=/bin/su"'

**Phase 2: Completed decoding.
       decoder: 'sudo'
       srcuser: 'manager'
       tty: 'pts/0'
       pwd: '/home/manager'
       dstuser: 'root'
       command: '/bin/su"'

**Phase 3: Completed filtering (rules).
       Rule id: '5402'
       Level: '3'
       Description: 'Successful sudo to ROOT executed'
**Alert to be generated.

```

Kind regards,

Alfonso Ruiz-Bravo

### CODE REVIEW

**Comment by rossengeorgiev** [comment]
Hi @Lopuiz, your base branch switch included an extra commit to this PR. 

image

The original commit only added a new rule without a problem.

**Comment by Lopuiz** [comment]
Hi @rossengeorgiev 

Thanks for letting me know.
I'm learning Git and I didn't realize the mistake.

Regards

**Comment by rossengeorgiev** [comment]
@Lopuiz, no worries. Keep at it 👍
You can also squash merge this to avoid including the 2 extra commits.

File: rules/0020-syslog_rules.xml [hunk 1]
```diff
@@ -523,6 +523,13 @@
     <group>pci_dss_10.2.5,pci_dss_10.2.2,gpg13_7.6,gpg13_7.8,gpg13_7.13,gdpr_IV_32.2,hipaa_164.312.b,nist_800_53_AU.14,nist_800_53_AC.7,nist_800_53_AC.6,</group>
   </rule>
 
+  <rule id="5407" level="3">
+    <if_sid>5400</if_sid>
+    <regex> ; USER=\S+ ; COMMAND=| ; USER=\S+ ; TSID=\S+ ; COMMAND=</regex>
+    <description>Successful sudo executed </description>
```
Comment (chemamartinez) [inline rules/0020-syslog_rules.xml]: Remove the space at the end of the description. The sentence should finish with a dot.

**Pull request merged**